### PR TITLE
BUGFIX: SD-520 Disable 'Stack to 100%' with line+column on left axis

### PR DESCRIPTION
**Pull request by naibinh**
---

[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [x] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [x] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [x] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards) (if applicable)
- [x] Successful `extended test - examples`
- [x] Successful `extended test - storybook`
- [x] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)
- [x] Squash commits

# Related PRs
- gdc-analytical-designer: https://github.com/gooddata/gdc-analytical-designer/pull/2334
- gdc-dashboards: https://github.com/gooddata/gdc-dashboards/pull/2104

# Related Jira tasks
- SD-520: https://jira.intgdc.com/browse/SD-520

### CODE REVIEW

**Comment by naibinh** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4145/ : SUCCESS in 7m 29s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5261/ : SUCCESS in 7s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2244/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1601/ : UNSTABLE in 13m 17s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5262/ : SUCCESS in 6s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4146/ : SUCCESS in 4m 18s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5263/ : SUCCESS in 7s

**Comment by naibinh** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4147/ : SUCCESS in 4m 22s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5264/ : SUCCESS in 6s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2245/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1602/ : UNSTABLE in 13m 15s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5265/ : SUCCESS in 9s

**Comment by naibinh** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4148/ : SUCCESS in 3m 59s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5266/ : SUCCESS in 8s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2246/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1603/ : SUCCESS in 10m 22s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5267/ : SUCCESS in 7s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4153/ : SUCCESS in 5m 39s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5277/ : SUCCESS in 7s

**Comment by naibinh** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4181/ : FAILURE in 4m 56s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5319/ : SUCCESS in 9s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2259/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1626/ : SUCCESS in 14m 13s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5320/ : SUCCESS in 8s

**Comment by naibinh** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1058.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4182/ : SUCCESS in 21m 51s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5321/ : SUCCESS in 14s

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1627/ : SUCCESS in 27m 53s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5322/ : SUCCESS in 7s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4206/ : SUCCESS in 4m 41s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5361/ : SUCCESS in 7s

File: src/components/visualizations/chart/test/chartOptionsBuilder.spec.ts [hunk 1]
```diff
@@ -3110,7 +3117,7 @@ describe("chartOptionsBuilder", () => {
         });
 
         describe("in usecase of combo chart", () => {
-            it("should assign `line` type to second serie according mbObject", () => {
+            it("should assign `line` type to second series according mbObject", () => {
                 const chartOptions = generateChartOptions(fixtures.comboWithTwoMeasuresAndViewByAttribute, {
                     type: "combo",
```
Comment (ptquang86) [inline src/components/visualizations/chart/test/chartOptionsBuilder.spec.ts]: is it `VisualizationTypes.COMBO`?

File: src/components/visualizations/chart/test/chartOptionsBuilder.spec.ts [hunk 2]
```diff
@@ -3152,6 +3159,45 @@ describe("chartOptionsBuilder", () => {
...
                 expect(chartOptions.yAxes[0].format).toBe(expectedPercentageFormat);
             });
+
+            it.each([[null, false], [NORMAL_STACK, true]])(
+                "should return %s when column+line chart is single axis and 'Stack Measures' is %s",
+                (stackingValue: string, stackMeasures: boolean) => {
+                    const chartOptions = generateChartOptions(
+                        fixtures.comboWithTwoMeasuresAndViewByAttribute,
+                        {
+                            type: "combo",
```
Comment (ptquang86) [inline src/components/visualizations/chart/test/chartOptionsBuilder.spec.ts]: is it `VisualizationTypes.COMBO`?

File: src/components/visualizations/chart/test/chartOptionsBuilder.spec.ts [hunk 3]
```diff
@@ -3152,6 +3159,45 @@ describe("chartOptionsBuilder", () => {
...
+            it.each([
+                [PERCENT_STACK, { stackMeasuresToPercent: true }],
+                [NORMAL_STACK, { stackMeasures: true }],
+            ])(
+                "should return %s when column+line chart is dual axis",
+                (stacking: string, stackingConfig: IMeasuresStackConfig) => {
+                    const chartOptions = generateChartOptions(
+                        fixtures.comboWithTwoMeasuresAndViewByAttribute,
+                        {
+                            type: "combo",
```
Comment (ptquang86) [inline src/components/visualizations/chart/test/chartOptionsBuilder.spec.ts]: same with above

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: src/components/visualizations/chart/test/chartOptionsBuilder.spec.ts (hunk 3 above)
Comment (naibinh) [reply src/components/visualizations/chart/test/chartOptionsBuilder.spec.ts]: done

File: src/components/visualizations/chart/test/chartOptionsBuilder.spec.ts (hunk 2 above)
Comment (naibinh) [reply src/components/visualizations/chart/test/chartOptionsBuilder.spec.ts]: done

File: src/components/visualizations/chart/test/chartOptionsBuilder.spec.ts (hunk 1 above)
Comment (naibinh) [reply src/components/visualizations/chart/test/chartOptionsBuilder.spec.ts]: done

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4222/ : SUCCESS in 9m 19s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5376/ : SUCCESS in 10s

File: src/components/visualizations/chart/chartOptions/test/comboChartOptions.spec.ts [hunk 4]
```diff
@@ -36,3 +37,50 @@ describe("getComboChartSeries", () => {
...
+                    },
+                ],
+                PERCENT_STACK,
+            ),
+        ).toBe(PERCENT_STACK);
+    });
+
+    it.each([[NORMAL_STACK, true], [null, false]])(
+        "should return %s stack value when 'Stack Measures' config is %s",
+        (stackValue: string, stackMeasures: boolean) => {
```
Comment (ptquang86) [inline src/components/visualizations/chart/chartOptions/test/comboChartOptions.spec.ts]: `stackValue` can be `null`, using `stackValue: string` here is not correct

File: src/components/visualizations/chart/chartOptions/test/comboChartOptions.spec.ts [hunk 5]
```diff
@@ -36,3 +37,50 @@ describe("getComboChartSeries", () => {
...
+    });
+
+    it("should return default 'percent' stack value", () => {
+        expect(
+            getComboChartStackingConfig(
+                { stackMeasures: true },
+                [
+                    {
+                        yAxis: 0,
+                        type: "column",
```
Comment (ptquang86) [inline src/components/visualizations/chart/chartOptions/test/comboChartOptions.spec.ts]: Using the constant here would be better. 
Could you please apply it here and the similar places

File: src/components/visualizations/chart/highcharts/test/getOptionalStackingConfiguration.spec.ts [hunk 6]
```diff
@@ -104,18 +104,14 @@ describe("getOptionalStackingConfiguration", () => {
             },
         );
 
-        it.each([
-            ["normal", { stackMeasures: true }, { stackMeasuresToPercent: false }],
-            ["percent", { stackMeasuresToPercent: true }, { stackMeasuresToPercent: true }],
-        ])(
+        it.each([["normal", { stackMeasures: true }], ["percent", { stackMeasuresToPercent: true }]])(
```
Comment (ptquang86) [inline src/components/visualizations/chart/highcharts/test/getOptionalStackingConfiguration.spec.ts]: We have the constants for `percent` and `normal`, right?

File: src/components/visualizations/chart/highcharts/test/getOptionalStackingConfiguration.spec.ts [hunk 7]
```diff
@@ -166,9 +160,9 @@ describe("getOptionalStackingConfiguration", () => {
                     }),
                     ...Array(2).fill({
                         yAxis: 1, // secondary Y axis
-                        // stack is resetted for secondary Y axis
+                        // stack is reset for secondary Y axis
                         stack: null,
-                        // stacking on secondary Y axis is resetted when stackMeasures is false
+                        // stacking on secondary Y axis is reset when stackMeasures is false
                         stacking: "normal",
```
Comment (ptquang86) [inline src/components/visualizations/chart/highcharts/test/getOptionalStackingConfiguration.spec.ts]: same as above :)

File: src/components/visualizations/chart/test/chartOptionsBuilder.spec.ts [hunk 8]
```diff
@@ -3110,26 +3117,26 @@ describe("chartOptionsBuilder", () => {
         });
 
         describe("in usecase of combo chart", () => {
-            it("should assign `line` type to second serie according mbObject", () => {
+            it("should assign `line` type to second series according mbObject", () => {
                 const chartOptions = generateChartOptions(fixtures.comboWithTwoMeasuresAndViewByAttribute, {
-                    type: "combo",
+                    type: VisualizationTypes.COMBO,
```
Comment (ptquang86) [inline src/components/visualizations/chart/test/chartOptionsBuilder.spec.ts]: Cache this `VisualizationTypes....` stuff at the top of this file would save some type. 
What do you think?

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4225/ : SUCCESS in 5m 50s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5380/ : SUCCESS in 8s

**Comment by ptquang86** [comment]
LGTM

File: src/components/visualizations/chart/test/chartOptionsBuilder.spec.ts (hunk 8 above)
Comment (naibinh) [reply src/components/visualizations/chart/test/chartOptionsBuilder.spec.ts]: done

File: src/components/visualizations/chart/highcharts/test/getOptionalStackingConfiguration.spec.ts (hunk 7 above)
Comment (naibinh) [reply src/components/visualizations/chart/highcharts/test/getOptionalStackingConfiguration.spec.ts]: done

File: src/components/visualizations/chart/highcharts/test/getOptionalStackingConfiguration.spec.ts (hunk 6 above)
Comment (naibinh) [reply src/components/visualizations/chart/highcharts/test/getOptionalStackingConfiguration.spec.ts]: done

File: src/components/visualizations/chart/chartOptions/test/comboChartOptions.spec.ts (hunk 5 above)
Comment (naibinh) [reply src/components/visualizations/chart/chartOptions/test/comboChartOptions.spec.ts]: done

File: src/components/visualizations/chart/chartOptions/test/comboChartOptions.spec.ts (hunk 4 above)
Comment (naibinh) [reply src/components/visualizations/chart/chartOptions/test/comboChartOptions.spec.ts]: done

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2289/ : SUCCESS in 13m 04s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4251/ : SUCCESS in 8m 21s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1087/ : SUCCESS in 21s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/1310/ : SUCCESS in 28m 56s

**Pull request merged**